syntax = "proto3";

import "commondefine.proto";

package Micro.Future.Message.Business;

message NamedStringVector
{
string name = 1;
repeated string entry = 2;
}

message NamedIntVector
{
string name = 1;
repeated int32 entry = 2;
}

message NamedDoubleVector
{
string name = 1;
repeated double entry = 2;
}

message SimpleIntTable
{
DataHeader hearder = 1;
repeated NamedIntVector columns = 2;
}

message SimpleStringTable
{
DataHeader hearder = 1;
repeated NamedStringVector columns = 2;
}

message SimpleDoubleTable
{
DataHeader hearder = 1;
repeated NamedDoubleVector columns = 2;
}

message ComplexTable
{
DataHeader hearder = 1;
SimpleIntTable     intTable = 2;
SimpleDoubleTable  doubleTable = 3;
SimpleStringTable stringTable = 4;
ComplexTable	   nestTable = 5;
}